dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/DataSourceServiceImpl.java

```diff
@@ -139,7 +140,7 @@
         }
 
         //check name can use or not
-        if (!dataSource.getName().trim().equals(dataSource.getName()) && checkName(dataSource.getName())) {
+        if (!dataSource.getName().trim().equals(dataSourceParam.getName()) && checkDatasourceNameExist(dataSourceParam.getName())) {
```

Comment:
I found this possible bug while adding the unit test and fixed it. Please let me know if it is not a bug.